src/content/doc-surrealql/datamodel/idioms.mdx

```diff
@@ -0,0 +1,601 @@
+
+### Field Access
+
+Since SurrealDB is, at its core, a document database each record is stored on an underlying key-value store storage engine, with the ability to store arbitrary arrays, objects, and many other types of data.  you can access fields in a record using the dot notation. To access a field in an object, use a dot `.` followed by the field name. 
+
+This is mostly helpful when accessing fields within a record, but can also be used to access fields within an array.
+
+For example, using the `CREATE` statement to  a record into the `person` table:
+
+
+```surql title="Query"
+CREATE person CONTENT {
+    name: "John Doe",
+    age: 30,
+    address: {
+      city: "New York",
+      country: "USA"
+    }
+};
+```
+```json title="Response"
+{
+  "person": {
+    "name": "John Doe",
+    "age": 30,
+    "address": {
+      "city": "New York",
+      "country": "USA"
+    }
+  }
+}
+```
+
+To access the `city` field within the `address` object, you can use the following idiom:
+
+```surql title="Query"
+SELECT address.city FROM person;
+```
+
+```json title="Response"
+[
+  {
+    "address": {
+      "city": "New York"
+    }
+  }
+]
+```
+In this example, `person.name` is an idiom that accesses the `name` field of the `person` record.
+
+### Index Access
+
+To access an element in an array by its index, use square brackets `[]` with the index inside. For example let's say we have a `school` record with some student results. 
+
+```surql title="Query"
+CREATE student SET results = [
+	{ score: 76, date: "2017-06-18T08:00:00Z", name: "Algorithmics" },
+	{ score: 83, date: "2018-03-21T08:00:00Z", name: "Concurrent Programming" },
+	{ score: 69, date: "2018-09-17T08:00:00Z", name: "Advanced Computer Science 101" },
+	{ score: 73, date: "2019-04-20T08:00:00Z", name: "Distributed Databases" },
+];
+```
+
+```json title="Response"
+[
+	{
+		id: student:urxaykt4qkbr8rs2o68j,
+		results: [
+			{
+				date: '2017-06-18T08:00:00Z',
+				name: 'Algorithmics',
+				score: 76
+			},
+			{
+				date: '2018-03-21T08:00:00Z',
+				name: 'Concurrent Programming',
+				score: 83
+			},
+			{
+				date: '2018-09-17T08:00:00Z',
+				name: 'Advanced Computer Science 101',
+				score: 69
+			},
+			{
+				date: '2019-04-20T08:00:00Z',
+				name: 'Distributed Databases',
+				score: 73
+			}
+		]
+	}
+]
+```
+
+To access the first student in the `results` array, you can use the following idiom:
+
+
+```surql
+SELECT results[0].score FROM student;
+```
+
+```json title="Response"
+[
+  {
+    "results": [
+      { "score": 76 }
+    ]
+  }
+]
+```
+
+Here, `results[0].score` accesses the score of the first student in the `results` array. 
+
+### All Elements
+
+To access all elements in an array or all fields in an object, use `[*]` or simply `*`. This is useful when you want to access all the elements in an array or all the fields in an object. 
```

Comment:
This one is interesting because accessing all is default behaviour and doesn't necessarily need the `[*]` (I think? Unless there is an example that actually needs it but can't think of anything), so SELECT results.score will return the same thing.

However, `[*]` is good to know because users will still see it like in this case:

```
DEFINE FIELD data ON TABLE stuff TYPE array<string>;
INFO FOR TABLE stuff;
```

Output:

```
	fields: {
		data: 'DEFINE FIELD data ON stuff TYPE array<string> PERMISSIONS FULL',
		"data[*]": 'DEFINE FIELD data[*] ON stuff TYPE string PERMISSIONS FULL'
	},
```

So maybe after the current example you could mention that `[*]` doesn't get used all that often due to being the default but will show up from time to time such as in the INFO statement above.

Reply:
I'll prefer to leave the example